test/prototype/test_parameter.py

```diff
@@ -0,0 +1,169 @@
+
+
+class TestMoE3d(nn.Module):
+    def __init__(self, num_experts: int, input_size: int, output_size: int) -> None:
+        """
+        This implementation is taken from:
+        https://github.com/huggingface/transformers/blob/6cade29278c4aee3f174f8950f97a3873bdb212f/src/transformers/models/jetmoe/modeling_jetmoe.py#L141
+        
+        Args:
+            num_experts (int):
+                Number of experts.
+            input_size (int):
+                Size of the input.
+            output_size (int):
+                Size of the output.
+        """
+        super().__init__()
+        self.weight = nn.Parameter(torch.randn(num_experts, output_size, input_size))
+        self.num_experts = num_experts
+        self.input_size = input_size
+        self.output_size = output_size
+
+    def forward(self, inputs, expert_size):
+        """
+        Forward pass of the JetMoeParallelExperts module.
+
+        Args:
+            inputs (Tensor):
+                Input tensor.
+            expert_size:
+                Expert size information.
+
+        Returns:
+            Tensor: Output tensor.
+        """
+        # return True
+        input_list = inputs.split(expert_size, dim=0)
+        output_list = []
+        for i in range(self.num_experts):
+            output_list.append(F.linear(input_list[i], self.weight[i]))
+        results = torch.cat(output_list, dim=0)
+        return results
+
+def print_model_fqn(model):
+    print("=== Parameters ===")
+    for name, param in model.named_parameters():
+        print(f"{name}: {param.shape}")
+    
+    print("\n=== Modules ===")
+    for name, module in model.named_modules():
+        if name:  # Skip empty name for root module
+            print(f"{name}: {type(module).__name__}")
+
+
+class TestQuantizeParameterMoE(common_utils.TestCase):
+
+    def test_2d_3d_moe_equivalent(self):
+        test_input = torch.randn(1024, 1024).cuda()
+
+        model_2d = TestMoE2d(2, 1024, 1024).cuda()
+        model_3d = TestMoE3d(2, 1024, 1024).cuda()
+
+        for i, expert in enumerate(model_2d.experts):
+            model_3d.weight.data[i] = expert.weight.detach().clone()
+
+        output_2d = model_2d(test_input, 512)
+        output_3d = model_3d(test_input, 512)
+        
+        torch.testing.assert_close(output_2d, output_3d, rtol=1e-3, atol=1e-3)
+
+
+    @unittest.skipIf(not torch.cuda.is_available(), "Need CUDA available")
+    def test_quantize_parameter(self):
+        test_input = torch.randn(1024, 1024).cuda().bfloat16()
+
+        model_2d = TestMoE2d(2, 1024, 1024).cuda().bfloat16()
+        model_3d = TestMoE3d(2, 1024, 1024).cuda().bfloat16()
+
+        for i, expert in enumerate(model_2d.experts):
+            model_3d.weight.data[i] = expert.weight.detach().clone()
+
+        # quantize all linears in 2d
+        quantize_(
+            model_2d,
+            Float8DynamicActivationFloat8WeightConfig(granularity=PerRow()),
+        )
+        
+        quantize_(
```

Comment:
this should target individual 3d params by name of the param, will probably need updates to `quantize_`.